app/app.py

```diff
@@ -58,77 +70,107 @@ def get_query_geom(coords):
         raise exceptions.ServerError("Insufficient xy coordinates provided. A LinearRing must have at least 3 coordinate tuples.", status_code=500)
         
 
-def get_intersect_features(query_geom, grid):
+def get_intersect_features(query_geom, grid, idx):
+    # get the grid cells that intersect with the request geometry
+    # see: https://stackoverflow.com/questions/14697442/faster-way-of-polygon-intersection-with-shapely
     ids = []
     polys = []
 
-    for feature in grid["features"]:
-        poly = shape(feature["geometry"])
+    if isinstance(query_geom, polygon.PolygonAdapter):
+        coords = query_geom.exterior.coords
+    else:
+        coords = query_geom.coords
+
+    # reduce intersection feature set with rtree (this tests polygon bbox intersection)
+    for intersect_pos in idx.intersection(query_geom.bounds):
 
+        grid_id = list(grid["FeatureIndex"].keys())[intersect_pos]
+        poly = shape(grid["FeatureIndex"][grid_id]["geometry"])
+        
+        # check if poly actually interesects with request geom
         if query_geom.intersects(poly):
-            ids.append(feature["properties"]["id"])
+            ids.append(grid["FeatureIndex"][grid_id]["properties"]["id"])
             polys.append(poly)
 
     return ids, polys
 
 
-def get_trip_features(source_ids, grid, flow):
-    # used to collect trip count values from grid cells that match a source cell
+def get_trip_features(intersect_ids, grid, flow, mode):
+    '''
+    Given a list of cell ids, extract trip count properties from the source grid data.
+    '''    
```

Comment:
what does this syntax do? Is it for documentation?

Reply:
@mateoclarke triple quoting is a [notorious](https://stackoverflow.com/questions/7696924/way-to-create-multiline-comments-in-python) python hack for multi-line comments that has been [widely adopted](https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings)

Reply:
...and, yes, by following specific "docstring" standards one can generate docs with tools like sphinx: https://thomas-cokelaer.info/tutorials/sphinx/docstring_python.html